pinot-core/src/main/java/org/apache/pinot/core/operator/combine/function/CombineFunction.java

```diff
@@ -0,0 +1,32 @@
+package org.apache.pinot.core.operator.combine.function;
+
+import org.apache.pinot.core.operator.blocks.results.BaseResultsBlock;
+
+
+public interface CombineFunction<T extends BaseResultsBlock> {
+
+  /**
+   * Merges a results block into the main mergeable results block.
+   * <p>NOTE: {@code blockToMerge} should contain the result for a segment without any exception. The errored segment
```

Comment:
this note is a little confusing, are you suggesting that `blockToMerge` should not have an exception so that the implementor of this method does not need to worry about checking whether or not the block has an exception in the metadata?

Reply:
note was copied but yeah I added clarification